x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/logic/result_settings_logic.ts

```diff
@@ -0,0 +1,200 @@
+        schemaConflicts,
+        ...splitResultFields(resultFields, schema),
+      };
+    },
+    clearAllFields: () => true,
+    resetAllFields: () => true,
+    updateField: (fieldName, settings) => ({ fieldName, settings }),
+    saving: () => true,
+  }),
+  reducers: () => ({
+    dataLoading: [
+      true,
+      {
+        initializeResultFields: () => false,
+      },
+    ],
+    saving: [
+      false,
+      {
+        initializeResultFields: () => false,
+        saving: () => true,
+      },
+    ],
+    openModal: [
+      OpenModal.None,
+      {
+        initializeResultFields: (_, { resultFields }: { resultFields: FieldResultSettingObject }) =>
+          areFieldsAtDefaultSettings(resultFields) ? OpenModal.ConfirmModifyModal : OpenModal.None,
+        closeModals: () => OpenModal.None,
+        resetAllFields: () => OpenModal.None,
+        openConfirmResetModal: () => OpenModal.ConfirmResetModal,
+        openConfirmSaveModal: () => OpenModal.ConfirmSaveModal,
+        saving: () => OpenModal.None,
+      },
+    ],
+    nonTextResultFields: [
+      {},
+      {
+        initializeResultFields: (
+          _,
+          { nonTextResultFields }: { nonTextResultFields: FieldResultSettingObject }
+        ) => nonTextResultFields,
+        clearAllFields: (nonTextResultFields) => clearAllFields(nonTextResultFields),
+        resetAllFields: (nonTextResultFields) => resetAllFields(nonTextResultFields),
+        updateField: (nonTextResultFields, { fieldName, settings }) =>
+          fieldName in nonTextResultFields
+            ? { ...nonTextResultFields, [fieldName]: settings }
+            : nonTextResultFields,
+      },
+    ],
+    textResultFields: [
+      {},
+      {
+        initializeResultFields: (
+          _,
+          { textResultFields }: { textResultFields: FieldResultSettingObject }
+        ) => textResultFields,
+        clearAllFields: (textResultFields) => clearAllFields(textResultFields),
+        resetAllFields: (textResultFields) => resetAllFields(textResultFields),
+        updateField: (textResultFields, { fieldName, settings }) =>
+          fieldName in textResultFields
+            ? { ...textResultFields, [fieldName]: settings }
+            : textResultFields,
+      },
+    ],
+    resultFields: [
+      {},
+      {
+        initializeResultFields: (_, { resultFields }: { resultFields: FieldResultSettingObject }) =>
+          resultFields,
+        clearAllFields: (resultFields) => clearAllFields(resultFields),
+        resetAllFields: (resultFields) => resetAllFields(resultFields),
+        updateField: (resultFields, { fieldName, settings }) =>
+          fieldName in resultFields ? { ...resultFields, [fieldName]: settings } : resultFields,
+      },
+    ],
+    serverResultFields: [
+      {},
+      {
+        initializeResultFields: (
+          _,
+          { serverResultFields }: { serverResultFields: ServerFieldResultSettingObject }
+        ) => serverResultFields,
```

Comment:
Same as above re: typing
```suggestion
        initializeResultFields: (_, { serverResultFields }) => serverResultFields,
```

Reply:
8646d41796cef61efa7abcde8f742ab1df4dbad8